### PR TITLE
Matches and foreach

### PR DESCRIPTION
**Pull request by RuuddR**
Added the new matches method and replaced the loop with the suggested line from the code review

### CODE REVIEW

File: Bootstrap.cs [hunk 1]
```diff
@@ -76,28 +77,41 @@ private void GetAnswer(string clipboardText)
...
+                //        {
+                //            ShowNotification(actionResult);
+                //            Clipboard.Clear();
+                //        }
+                //        break;
+                //    }
+
+                //}
+
+                var service = serviceProvider.GetServices<IAction>().FirstOrDefault(s => s.Matches(clipboardText));
```
Comment (RyanThiele) [inline Bootstrap.cs]: Slick, only using the services (dependencies) needed. This will speed up the processes.

**Review by RyanThiele — COMMENTED** [review]
@M1sterPerfect From what I gather, you have added all available commands to an read only array. The matches are then used to filter the services in bootstrap.cs. Since the matches are used as a filter, would it be better to name the read only property `Filter`, or should you keep it as `Matches`?

**Comment by RyanThiele** [comment]
BTW good rework

**Comment by TeunvandenBroek** [comment]
It already looks good! Should anything else happen to it before the pull request is accepted?

**Comment by TeunvandenBroek** [comment]
@M1sterPerfect @RyanThiele  only my questions class isnt working now.

**Comment by RyanThiele** [comment]
Created a new PR and corrected issues

**Pull request merged**